### PR TITLE
Upgrade mockito

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR upgrades the mockito version and removes powermockito from `utils` module.

Moving to Mockito to mock static methods (available since 3.4.0) instead of powermockito since it hasn't had any release since 2020 and is not compatible with newer versions of mockito.

I will create separate PRs for other modules overtime and eventually remove powermockito.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by harikrishna-patnala** [comment]
@blueorangutan package

File: pom.xml [hunk 1]
```diff
@@ -113,7 +113,7 @@
         <cs.junit.dataprovider.version>1.13.1</cs.junit.dataprovider.version>
         <cs.junit.jupiter.version>5.9.1</cs.junit.jupiter.version>
         <cs.guava-testlib.version>18.0</cs.guava-testlib.version>
-        <cs.mockito.version>3.2.4</cs.mockito.version>
```
Comment (harikrishna-patnala) [inline pom.xml:116]: we may also need the following change, so mockstatic() can be used with Mockito itself.

-      <artifactId>mockito-core</artifactId>
+      <artifactId>mockito-inline</artifactId>

Let's first run tests and see nothing is broken

Comment (DaanHoogland) [reply pom.xml:116]: @harikrishna-patnala tests have been run in the github acrions. Do you need to see anything else?

Comment (vishesh92) [reply pom.xml:116]: @harikrishna-patnala Let me make this change. Then we can incrementally remove PowerMockito.

File: utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 2]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (DaanHoogland) [inline utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: license?

Comment (vishesh92) [inline utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Need to make this change to fix this error.
ref: https://javadoc.io/doc/org.mockito/mockito-core/3.12.4/org/mockito/Mockito.html#Mocking_Final
```
org.mockito.exceptions.base.MockitoException: 
The used MockMaker PowerMockMaker does not support the creation of static mocks

Mockito's inline mock maker supports static mocks based on the Instrumentation API.
You can simply enable this mock mode, by placing the 'mockito-inline' artifact where you are currently using 'mockito-core'.
Note that Mockito's inline mock maker is not supported on Android.
```

Comment (DaanHoogland) [reply utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:17]: We can add it to the excluded files in pom.xml as well @vishesh92 (if that is the wiser thing to do?)

Comment (vishesh92) [reply utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:17]: I don't have much experience with `pom.xml`. Are you suggesting that we add exclusion criteria to `powermock` dependencies in `pom.xml`

Comment (DaanHoogland) [reply utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:17]: no , just an exclusion to the license requitrement for this config file that only contains code that we did not come up with ;)
in pom.xml at project level around line 1064 add
```
                            <exclude>utils/src/test/resources/mockito-extension/org.mockito.plugins.MockMaker</exclude>
```

Comment (vishesh92) [reply utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Got it. I have updated the PR.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Review by rohityadavcloud — APPROVED** [review]

File: utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 2 above)
Comment (rohityadavcloud) [inline utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: @vishesh92 is this boilerplate or does it serve any purpose? If you want an empty file you can touch the file and git add.

**Comment by rohityadavcloud** [comment]
@blueorangutan package

File: utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 2 above)
Comment (vishesh92) [reply utils/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: We will need to add this file to module when we remove powermock. It specifies the MockMaker to use in that module for running tests.

more info here: https://github.com/apache/cloudstack/pull/7577/files#r1217685958

**Pull request merged**